woocommerce/woocommerce-ios
# Update Order to use OrderStatusEnum

# Why

While working on #2848 I noticed that the `Order` model was using a `String` to represent the order status. This PR updates the `Order` model to use the existing `OrderStatusEnum`

# Testing Steps
- Make sure tests pass.
- Do a smoke test around orders and updating orders to and verify that everything is working

Update release notes:
